core/src/raw/http_util/bytes_range.rs

```diff
@@ -77,15 +77,15 @@ impl BytesRange {
     ///
     /// Panic if input `n` is larger than the size of the range.
     pub fn advance(&mut self, n: u64) {
-        self.0 += n;
-        self.1 = self.1.map(|size| size - n);
+        self.offset += n;
+        self.size -= n;
     }
 
     /// Check if this range is full of this content.
     ///
     /// If this range is full, we don't need to specify it in http request.
     pub fn is_full(&self) -> bool {
-        self.0 == 0 && self.1.is_none()
+        self.offset == 0 && self.size == 0
```

Comment:
This is wrong. And we should remove `is_full` API.

Reply:
Yes. I first change it to `false` and will further refactor it.